sql-plugin/src/main/spark340/scala/com/nvidia/spark/rapids/shims/CreateDataSourceTableAsSelectCommandMetaShims.scala

```diff
@@ -48,7 +48,8 @@ final class CreateDataSourceTableAsSelectCommandMeta(
 
     val spark = SparkSession.active
     origProvider =
-      GpuDataSource.lookupDataSourceWithFallback(cmd.table.provider.get, spark.sessionState.conf)
+      GpuDataSourceBase.lookupDataSourceWithFallback(cmd.table.provider.get,
+        spark.sessionState.conf)
     // Note that the data source V2 always fallsback to the V1 currently.
     // If that changes then this will start failing because we don't have a mapping.
     gpuProvider = origProvider.getConstructor().newInstance() match {
```

Comment:
To be honest, GpuFileFormat is not really necessary in the GpuDataSource shim for 340. Since the command ultimately returned is a regular InsertIntoHadoopFsRelationCommand, the GPU checks are performed twice now. But I haven't removed this yet, just in case there is a dependency that I have missed.